common/docserver/docserver.go

```diff
@@ -95,14 +95,20 @@ func (self *DocServer) Get(uri, path string) (content []byte, err error) {
 			resp.Body.Close()
 		}
 	}()
+
 	if err != nil {
 		return
 	}
+
 	content, err = ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return
 	}
 
+	if resp.StatusCode >= 400 {
```

Comment:
You can reverse the check and test if you got a valid response (status code 200) instead of testing for an error. `http.Client.Get` handles most redirects so you should in almost all normal cases get a valid response.